### PR TITLE
Enhance and fix some ImageGeographicLocation elements and add tests f…

### PR DESCRIPTION
**Pull request by deanlinsalata**
Enhance and fix some ImageGeographicLocation elements and add tests for all ImageGeographicLocations

### CODE REVIEW

File: src/test/scala/com/tresys/nitf/TestNTBBaseline.scala [hunk 1]
```diff
@@ -23,6 +23,11 @@ class TestNTBBaseline {
   @Test def test_i_3025b() { runner.runOneTest("i_3025b") }
   @Test def test_i_3034c() { runner.runOneTest("i_3034c") }
   @Test def test_i_3034f() { runner.runOneTest("i_3034f") }
+  @Test def test_i_3034f() { runner.runOneTest("i_3034f_decimal") }
+  @Test def test_i_3034f() { runner.runOneTest("i_3034f_geographic") }
+  @Test def test_i_3034f() { runner.runOneTest("i_3034f_military") }
+  @Test def test_i_3034f() { runner.runOneTest("i_3034f_utm_north") }
+  @Test def test_i_3034f() { runner.runOneTest("i_3034f_utm_south") }
```
Comment (stevedlawrence) [inline src/test/scala/com/tresys/nitf/TestNTBBaseline.scala:30]: These def's all have the same name so there are duplicate method error when trying to run the tests. These should be named to match the test name (e.g. ``def test_i_3034f_decimal()``)

File: src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd [hunk 2]
```diff
@@ -480,8 +486,26 @@ The message root is 'NITF'.
                                 <xs:element name="Corner" minOccurs="4" maxOccurs="4" dfdl:occursCountKind="fixed">
                                   <xs:complexType>
                                     <xs:sequence>
-                                      <xs:element name="Latitude" type="nitf:BCS-A" dfdl:length="7" />
-                                      <xs:element name="Longitude" type="nitf:BCS-A" dfdl:length="8" />
+                                      <xs:element name="Latitude">
```
Comment (stevedlawrence) [inline src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd:489]: I definitely like this change, but a few of the existing tests will need to be updated to expect the new Degrees/Minutes/Seconds/NorthSouth. Make sure to run sbt test to make sure they've all be fixed.

File: src/test/scala/com/tresys/nitf/TestNTBBaseline.scala (hunk 1 above)
Comment (deanlinsalata) [reply src/test/scala/com/tresys/nitf/TestNTBBaseline.scala:30]: fixing now...

**Comment by deanlinsalata** [comment]
In testing, I have been short circuiting the jpeg schema in the dfdl-nitf schema, not to get held up in what I was trying to accomplish. There were many warnings in the jpeg schema and some also in the dfdl-nitf schema. I eliminated all warnigs in the dfdl-nitf schema but did not commit them here because I was not sure I was doing the right thing without extensive knowledge on how the whole schema was built. Below is what was showing for warnings. I fixed a couple of casts and took out the ignored attributes.

[info] Compiler: component counts: unique 368, actual 496.
[warning] Schema Definition Warning: In expression fn:ceiling(../PadOutputPixelCodeLength div 8.0), result type (Double) should be manually cast to the expected type (Long) with the appropriate constructor.Performing deprecated automatic conversion.
Schema context: PadOutputPixelCode Location line 637 column 28 in file:/home/clipdev/Desktop/tresys_dfdl/NITF/src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd
[warning] Schema Definition Warning: In expression ../../../Header/NumberOfBitsPerPixelPerBand, result type (UnsignedInt) should be manually cast to the expected type (NonEmptyString) with the appropriate constructor.Performing deprecated automatic conversion.
Schema context: choice[3] Location line 644 column 36 in file:/home/clipdev/Desktop/tresys_dfdl/NITF/src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd
[warning] Schema Definition Warning: DFDL property was ignored: representation="binary"
Schema context: Data Location line 748 column 16 in file:/home/clipdev/Desktop/tresys_dfdl/NITF/src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd
[warning] Schema Definition Warning: DFDL property was ignored: representation="text"
Schema context: Data Location line 781 column 16 in file:/home/clipdev/Desktop/tresys_dfdl/NITF/src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd
[warning] Schema Definition Warning: DFDL property was ignored: representation="binary"
Schema context: DESUserDefinedData Location line 815 column 22 in file:/home/clipdev/Desktop/tresys_dfdl/NITF/src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd
[warning] Schema Definition Warning: DFDL property was ignored: representation="binary"
Schema context: RESUserDefinedData Location line 843 column 22 in file:/home/clipdev/Desktop/tresys_dfdl/NITF/src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd

**Comment by stevedlawrence** [comment]
It looks like the following tests fail:
```
test_i_3034f_utm_south
test_i_3034f_utm_north
test_i_3034f_military
test_i_3034f_geographic
test_i_3034f_decimal
```
In each of these cases, it looks like the only difference is the header length (``NITF/Header/ImageSegmentLengths/HeaderLength``) in the expected xml files is 450, but the actual length is 510. That's a difference of 60, which I think makes sense since ``ImageGeograpicRepresentation`` takes up 60 bytes, and the original ``test_i_3034f`` file didn't have that, so that's 60 bytes added. Can you please update the expected infoset to have the 510 length?

Once you've done that, can you then please squash your commits insto a single commit using ``git rebase`` and then I think this should be good to merge!

**Comment by stevedlawrence** [comment]
Regarding the warnings, adding the cass and removing the properties is the correct thing to do to fix them. It would probably make sense to create separate pull request to fix those since they are unrelated to this change of fixing the geographic representation. Thanks!

**Comment by deanlinsalata** [comment]
I am new to working collaboratively on githup and never did a rebase w/squash but I think I have the single commit with all changes rolled into that commit, which makes a lot sense for a merge. Just a note, I did manually dfdl parse every test ntf file through the edited schema, last Friday, to a remote location other then the test directory, and all worked. I also unparsed my new xml test files to new ntf files and all went well even with the wrong header lengths which produced good header lengths in the ntf files. Sorry for the ignorance on SBT, I am not familiar with the tool, it was not installed on my machine and when I did install SBT there were some dependency problems that looked hard to fix since Mitre schema repository looked like it moved. Thanks with the help with the testing.

**Comment by stevedlawrence** [comment]
Rebase and changes look good and all test pass--good to merge. Thanks for the contribution!

**Pull request merged**